tasklist/client/src/Tasks/Task/Details/Header.tsx

```diff
@@ -137,11 +138,50 @@ const AssignButton: React.FC<{
         tracking.track({eventName: 'task-assigned'});
       }
     } catch (error) {
-      const errorMessage = (error as Error).message ?? '';
+      const errorMessage = parseJSON((error as Error).message);
+      if (
+        errorMessage?.status === 504 &&
```

Comment:
We shouldn't return the status code in the body, it's already present in the response'
This logic should be encapsulated on the backend, we should be able to derive the meaning/message from the error only by the error code